File: src/main/java/io/github/oleksivio/telegram/bot/api/model/method/update/GetWebhookInfo.java
package io.github.oleksivio.telegram.bot.api.model.method.update;

import io.github.oleksivio.telegram.bot.api.model.NetworkError;
import io.github.oleksivio.telegram.bot.core.controller.network.ActionNetworker;
import io.github.oleksivio.telegram.bot.core.model.method.RunnableAction;
import io.github.oleksivio.telegram.bot.core.model.responses.CommonResponse;
import io.github.oleksivio.telegram.bot.core.model.responses.WebhookInfoResponse;
import io.github.oleksivio.telegram.bot.api.model.objects.WebhookInfo;

/**
 * @see <a href="https://core.telegram.org/bots/api#getwebhookinfo">getWebhookInfo</a>
 */
public class GetWebhookInfo extends RunnableAction<WebhookInfo> {
    private static final String METHOD = "getWebhookInfo";

    public GetWebhookInfo(ActionNetworker actionNetworker) {
        super(METHOD, actionNetworker);
    }

    @Override
    protected Class<? extends CommonResponse<WebhookInfo>> getResultWrapperClass() {
        return WebhookInfoResponse.class;
    }

    @Override
    public GetWebhookInfo setNetworkErrorListener(NetworkError onNetworkError) {
        pSetNetworkErrorListener(onNetworkError);
        return this;
    }
}

File: src/main/kotlin/io/github/oleksivio/telegram/bot/api/model/method/update/GetWebhookInfo.kt
package io.github.oleksivio.telegram.bot.api.model.method.update

import com.fasterxml.jackson.annotation.JsonIgnore
import com.fasterxml.jackson.annotation.JsonProperty
import io.github.oleksivio.telegram.bot.api.model.objects.WebhookInfo
import io.github.oleksivio.telegram.bot.core.model.ApiDict
import io.github.oleksivio.telegram.bot.core.model.WebhookInfoResponse
import io.github.oleksivio.telegram.bot.core.model.method.Action

/**
 * @see [getWebhookInfo](https://core.telegram.org/bots/api/#getwebhookinfo)
 */
class GetWebhookInfo : Action<WebhookInfo>() {
    @JsonProperty(ApiDict.METHOD_KEY)
    override val method: String = "getWebhookInfo"

    @JsonIgnore
    override val resultWrapperClass = WebhookInfoResponse::class

}
